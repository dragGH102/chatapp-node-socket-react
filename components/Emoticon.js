import PropTypes from 'prop-types';
import FontAwesome from 'react-fontawesome';

const Emoticon = ({ name, className }) => (<FontAwesome
    name={name}
    className={className}
/>;

Emoticon.propTYpes = {
    className: PropTypes.string,
    name: PropTypes.string.isRequired,
};

export default Emoticon;