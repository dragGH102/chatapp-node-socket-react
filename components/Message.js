import MessageType from '../types/Message';

const Message = ({ message }) => (<div>
  <span
    className={`message ${message.author === 'me' ? 'mine' : 'other'}`}
    style={message.css}
  >
    {message.content}
  </span>
  <style jsx>{`
    .message {
      width: 40%;
      clear: both;
      border: 1px solid #000;
      border-radius: 5px;
      height: 1.5em;
      font-size: 1em;
      padding: 2px 10px;
      margin-top: 10px;
    }

    .message.mine {
      float: right;
      text-align: right;
      background-color: #98FB98;
    }

    .message.other {
      float: left;
      text-align: left;
      background-color: #D3D3D3;
    }
  `
  }</style>
</div>);

Message.propTypes = {
  message: MessageType.isRequired,
};

export default Message;
