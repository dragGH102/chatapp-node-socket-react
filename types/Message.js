import PropTypes from 'prop-types';

export default PropTypes.shape({
  id: PropTypes.number.isRequired,
  author: PropTypes.string.isRequired, // me | other
  content: PropTypes.string.isRequired,
  css: PropTypes.string,
  type: PropTypes.string, // <command> | null
  sent: PropTypes.bool,
});
