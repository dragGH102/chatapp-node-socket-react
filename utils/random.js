module.exports = {
    randomNumberWithingRange(min, max) {
        return Math.floor(Math.random() * (max - min + 1)) + min;
    }
};