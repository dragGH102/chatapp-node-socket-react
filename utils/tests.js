import randomUtils from "./random";

module.exports = {
    generateMessage(author) {
        return {
            id: new Date().getUTCMilliseconds() + randomUtils.randomNumberWithingRange(1, 1000),
            content: 'anything',
            author,
            css: {},
        };
    }
};