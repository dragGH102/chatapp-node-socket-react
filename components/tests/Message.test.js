import React from 'react'
import renderer from 'react-test-renderer'
import Message from "../Message";
import { generateMessage } from "../../utils/tests";
import { mount } from "enzyme";
import Emoticon from "../Emoticon";

describe('<Message />', () => {
    it('should render correctly"', () => {
        // a similar test (snapshot test) can be written for the other components
        const message = generateMessage('me');

        const component = renderer.create(<Message message={message} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should include EMOTICON component if valid emoticon string provided', () => {
        const message = generateMessage('me');
        message.content = 'Hey, this is a (smile)! ... but now I am (frown)';

        const component = mount(<Message message={message} />);
        expect(component.find(Emoticon)).toHaveLength(2);
    });
});