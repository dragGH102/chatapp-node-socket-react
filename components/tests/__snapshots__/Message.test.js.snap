// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> should render correctly" 1`] = `
<div
  className="slide-in-right undefined"
  data-jsx={586277062}
>
  <span
    className="message mine"
    data-jsx={586277062}
    style={Object {}}
  >
    anything
  </span>
</div>
`;
